app/src/main/java/com/android/streetworkapp/ui/park/ParkOverview.kt

```diff
@@ -253,6 +282,59 @@ fun RatingDialog(showDialog: MutableState<Boolean>) {
   }
 }
 
+/**
+ * Verifies that the current state is correct and then rates the park.
+ *
+ * @param context The context of the application.
+ * @param park The park to rate.
+ * @param user The user who is rating the park.
+ * @param starRating The rating value.
+ * @param parkViewModel The park view model.
+ */
+fun handleRating(
+    context: Context?,
+    park: Park?,
+    user: User?,
+    starRating: Int,
+    parkViewModel: ParkViewModel?
+) {
+  Log.d("ParkOverview", "handleRating: {park=$park ; user=$user ; rating=$starRating")
+
+  if (user == null) {
```

Comment:
Maybre you can use kotlin when function instead of if else, i find it more readable.
https://www.educba.com/kotlin-when-multiple-conditions/

Reply:
Whens would not make it more redable in this case because the variable is never the same : 

`when(user){
  null -> {Toast}
  else -> {
    when(park){
      null -> {Toast}
      else -> {
        .....
      }
    }
}
}`

Would it be better for your readability to remove the else of the else if (and add `return@handleRating` inside each ifs) would make it better by allowing more space between each check by providing less nesting though ?

Reply:
sorry it was not a good link 
https://kotlinlang.org/docs/control-flow.html#when-expressions-and-statements
I meant somethin like this 

```
when {
    a > b -> "a is greater than b"
    a < b -> "a is less than b"
    else -> "a is equal to b"
}
```
it check if statement is true

Reply:
Ok ! I understand what you meant by when now ! I changed it :)

Reply:
It was purely for eshtetic if you find if-else more appropriate it can be changed